core/domain/collection_services.py

```diff
@@ -1036,6 +1038,15 @@ def compute_collection_contributors_summary(collection_id):
             break
 
         current_version -= 1
+
+    # Remove IDs that are deleted or do not exists.
+    users_settings = user_services.get_users_settings(
+        list(contributors_summary))
```

Comment:
Perhaps contributors_summary.keys() to be more explicit.

Or, define contributor_ids = list(contributors_summary) so you can use it here and below.